tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Provides a library to deal with the CoCo KBS
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+kubernetes_dir="$(dirname "$(readlink -f "$0")")"
+# shellcheck disable=1091
+source "${kubernetes_dir}/../../gha-run-k8s-common.sh"
+
+# Where the kbs sources will be cloned
+readonly COCO_KBS_DIR="/tmp/kbs"
+
+# Delete the kbs on Kubernetes
+#
+# Note: assume the kbs sources were cloned to $COCO_KBS_DIR
+#
+function kbs_k8s_delete() {
+	pushd "$COCO_KBS_DIR"
+	kubectl delete -k kbs/config/kubernetes/overlays
+	popd
+}
+
+# Deploy the kbs on Kubernetes
+#
+function kbs_k8s_deploy() {
+	local image
+	local image_tag
+	local ns="coco-tenant"
+	local repo
+	local kbs_ip
+	local kbs_port
+	local version
+
+	# yq is needed by get_from_kata_deps
+	ensure_yq
+
+	# Read from versions.yaml
+	repo=$(get_from_kata_deps "externals.coco-kbs.url")
+	version=$(get_from_kata_deps "externals.coco-kbs.version")
+	image=$(get_from_kata_deps "externals.coco-kbs.image")
+	image_tag=$(get_from_kata_deps "externals.coco-kbs.image_tag")
+
+	if [ -d "$COCO_KBS_DIR" ]; then
+		rm -rf "$COCO_KBS_DIR"
+	fi
+
+	echo "::group::Clone the kbs sources"
+	git clone --depth 1 "${repo}" "$COCO_KBS_DIR"
+	pushd "$COCO_KBS_DIR"
+	git fetch --depth=1 origin "${version}"
+	git checkout FETCH_HEAD -b kbs_$$
```

Comment:
Do we need the `PID` suffix if we make `COCO_KBS_DIR` unique?

Reply:
If we are going to store the kbs key here (and access it from the tests), we might not want this path to change.

Reply:
Good point @jodh-intel. In theory it can be just `kbs` or something well; and hope no one has the great idea to name a remote branch like that :)

@fitzthum this is just the temporary branch name. We can even checkout FETCH_HEAD without branching.

Reply:
This checks the main and then re-fetches the desired version, how about using:

```bash
git init $COCO_KBS_DIR
git remote add origin https://github.com/confidential-containers/kbs
git fetch --depth 1 origin "${version}"
git checkout FETCH_HEAD
```

(also I don't see a reason to branch the FETCH_HEAD, it should be sufficient to check it out, shouldn't it?)

Reply:
hi @ldoktor , check FETCH_HEAD out should be sufficient but I've this habit of branching to silence the `You are in 'detached HEAD' state` warning.